src/libpriv/rpmostree-util.c

```diff
@@ -226,29 +226,130 @@ rpmostree_translate_path_for_ostree (const char *path)
+ * prefix. increment_start specifies the version number to start at
+ * if a version number is not present in last_version.
+ * 
+ * If a cannot be parsed from last_version, or last_version
+ * does not begin with prefix, then append increment_start to prefix.
+ * 
+ * Example:
+ * last_version == NULL, prefix == 10, increment_start == 0 --> return 10.0
+ * 
+ * If both prefix and last_version are equal, then append increment_start.
+ * 
+ * Example:
+ * last_version == 10, prefix == 10, increment_start == 5 --> return 10.5
+ * 
+ * If increment_start is set to NULL, then append ".1"
+ * to prefix if and only if last_version is equal to prefix. 
+ * 
+ * Example:
+ * last_version == NULL, prefix == 10, increment_start == NULL --> return 10
+ * last_version == 10, prefix == 10, increment_start == NULL --> return 10.1
+ * 
+ * Returns the next version string.
+ */
+static char*
+increment_version (const char *last_version,
+                   const char *prefix,
+                   const char *increment_start)
 {
-  unsigned long long num = 0;
   const char *end = NULL;
+  unsigned long long num = 0;
+  const gboolean increment_given = increment_start != NULL;
+
+  g_assert_cmpstr (prefix, !=, NULL);
 
-  if (!last_version || !g_str_has_prefix (last_version, auto_version_prefix))
-    return g_strdup (auto_version_prefix);
+  if (!last_version || !g_str_has_prefix (last_version, prefix))
+    return increment_given ? g_strdup_printf ("%s.%s", prefix, increment_start)
+                           : g_strdup (prefix);
 
-  if (g_str_equal (last_version, auto_version_prefix))
-    return g_strdup_printf ("%s.1", auto_version_prefix);
+  if (g_str_equal (last_version, prefix))
+    return increment_given ? g_strdup_printf ("%s.%s", prefix, increment_start)
+                           : g_strdup_printf ("%s.1", prefix);
 
-  end = last_version + strlen(auto_version_prefix);
+  g_assert_cmpuint (strlen (last_version), >, strlen (prefix));
+  end = last_version + strlen (prefix);
 
   if (*end != '.')
-    return g_strdup (auto_version_prefix);
+    return increment_given ? g_strdup_printf ("%s.%s", prefix, increment_start)
+                           : g_strdup (prefix);
   ++end;
 
   num = g_ascii_strtoull (end, NULL, 10);
-  return g_strdup_printf ("%s.%llu", auto_version_prefix, num + 1);
+  return g_strdup_printf ("%s.%llu", prefix, num + 1);
 }
 
+#define VERSION_TAG_REGEX "<([a-zA-Z]+):(.*)?>"
+
+/* Get the next version, given a version prefix and a last version.
+ * Checks for supported version fields in the auto_version_prefix
+ * and renders them.
+ * 
+ * Returns the next version string if successful, NULL otherwise.
+ */
+char *
+_rpmostree_util_next_version (const char   *auto_version_prefix,
+                              const char   *last_version,
+                              GError      **error)
+{
+  bool date_tag_given = FALSE;
+  g_autoptr(GRegex) tag_regex = g_regex_new (VERSION_TAG_REGEX, 0, 0, error);
```

Comment:
Minor: we usually wrap regex compilation around a `g_once_init_enter()` (see other spots in this code). It's not as crucial here since we're only calling this once. But it's good practice for if we somehow refactor this code and it does end up being used multiple times. (Actually, the unit tests would benefit from this as well).

Reply:
Makes sense, thanks!